[Feature] Investigate dependabot package upgrade failure

### Search before asking

- [x] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.


### Description

A detailed failure case:
- A failed package upgrade: https://github.com/ray-project/kuberay/pull/3396
- A successful package upgrade: https://github.com/ray-project/kuberay/pull/3443

The only difference lies in golang and golang toolchain upgrade.
It would be nice (and somehow necessary) to investigate how to configure dependabot properly, otherwise we have to manually fix all the upgrade breakage ourselves.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

In #3396, the protobuf version for the apiserver was v1.36.5 whereas v1.36.6 in #3443.

While PR #3396 was for bumping protobuf version for `./proto`, `go mod download` in the workflow bumped up the version in the `./apiserver` as well (v1.36.5 ->  v1.36.6). However, `go mod download` didn't handle the protobuf version in the go.mod correctly which led to a build failure.

The version bump wasn't required in #3443, the breakage was avoided.
I can successfully build `apiserver` by running `go mod tidy` instead.

### Repro steps:
1. Checkout on commit `d40d1f6`
2. `cd apiserver`
3. `go mod download`
4. `go build ./...`, and the output is
```
% go mod download
% go build ./...
$HOME/go/pkg/mod/google.golang.org/grpc@v1.65.0/credentials/credentials.go:33:2: missing go.sum entry for module providing package google.golang.org/protobuf/proto (imported by github.com/ray-project/kuberay/apiserver/test/e2e); to add:
        go get github.com/ray-project/kuberay/apiserver/test/e2e
$HOME/go/pkg/mod/google.golang.org/grpc@v1.65.0/internal/pretty/pretty.go:27:2: missing go.sum entry for module providing package google.golang.org/protobuf/encoding/protojson (imported by github.com/ray-project/kuberay/apiserver/cmd); to add:
        go get github.com/ray-project/kuberay/apiserver/cmd
$HOME/go/pkg/mod/google.golang.org/grpc@v1.65.0/internal/pretty/pretty.go:28:2: missing go.sum entry for module providing package google.golang.org/protobuf/protoadapt (imported by google.golang.org/grpc/encoding/proto); to add:
        go get google.golang.org/grpc/encoding/proto@v1.65.0
$HOME/go/pkg/mod/google.golang.org/grpc@v1.65.0/binarylog/grpc_binarylog_v1/binarylog.pb.go:28:2: missing go.sum entry for module providing package google.golang.org/protobuf/reflect/protoreflect (imported by github.com/grpc-ecosystem/grpc-gateway/v2/runtime); to add:
        go get github.com/grpc-ecosystem/grpc-gateway/v2/runtime@v2.20.0
...
```

run `go mod tidy` instead:
3. `go mod tidy`
4. `go build ./...`
```
% go mod tidy
% go build ./...
%  # no error message
```

The difference was that protobuf version wasn't updated correctly in the go.mod

`go mod download`
![Image](https://github.com/user-attachments/assets/8a63a7b6-0e55-4535-8407-87faa23230b7)


`go mod tidy`
![Image](https://github.com/user-attachments/assets/a1e6f1be-418f-48b9-83df-95aa1337a6de)

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -15,3 +15,6 @@ updates:
       kubernetes:
         patterns:
           - "k8s.io/*"
+      google-golang:
+        patterns:
+          - "google.golang.org/*"

